### ISSUE TITLE
new retropie forum thread

### ISSUE THREAD

**Issue by markwkidd** [libretro/mame2003-plus-libretro#92]
Hello all. The retropie forum thread now is great, but I noticed that there are more than 460 posts! There is some misunderstandings about the relationship between the two cores which are based on the title of the thread and the early posts. Just one example of how it's out of date.

@Wilstorm or someone else, would you be willing to start a new thread with a title that includes the word "mame2003-plus"?

It could be anything. "mame2003-plus: brave new core (testers needed)", "mame2003-plus: brilliant advances and broken builds (testers needed)", "mame2003-plus: what does god need with a starship?"

Sometimes it's awkward for me to post a thread like this in the RetroPie forum when people realize I don't personally have/use RetroPie, just RetroArch.

If such a new thread exists, I'll immediately be on hand as with the existing thread.

**Comment by markwkidd** [comment]
One thing I'm thinking is that the core is probably ready for even more testing and testers. Especially with RetroPie 4.4 just out, this seems like a good moment to welcome a few more to the mystery.

**Comment by Wilstorm** [comment]
lol...personally I like what does god need with a starship! Something ummm...less majestic would be fine though too! I don't think anyone would mind that you're not using RetroPie. My only thought is if you're the thread starter it would allow any pertinent or important information to be posted in the first thread that can be updated or added later as needed. Or refer people to the top of the thread for useful information.

**Comment by grant2258** [comment]
@Wilstorm your great with people on the forum though you go out your way to help out ect not many do that

**Comment by Wilstorm** [comment]
I agree it's a runaway thread. I would be intimidated to sift hours through pages of information looking for an answer plus I am not sure what happened to darknoir (OP)...we've lost control of the thread!

**Comment by dankcushions** [comment]
could steal the formatting of my old mame2003 development thread: https://retropie.org.uk/forum/topic/2889/mame-development-lr-mame2003-updates :)

**Comment by Wilstorm** [comment]
@grant2258 that is very kind of you...was that a butter job? ;) To be honest I am a cheerleader on the sidelines rooting for my favorite team because you guys are saving me a LOT of quarters! Honestly honestly the work you guys do is incredible and I could never say thank you enough for the gift of retro gaming in my living room.

@dankcushions - I like that format and always found it incredibly helpful and easy to follow.

**Comment by Wilstorm** [comment]
@markwkidd - Hey Mark, you had one of those evenings getting up in the middle of the night to fix something? Having a eureka moment of clarity and the next day damn I'm tired! ;) If I get a solid 6 I am ok, If it's restless sleep a bit longer. 

I can start a thread if no one else will as I know you guys stay busy with programming. I think having anyone test anything or even their favorites is all beneficial to uncovering even minute errors. I've been working my way through my favorites list which is roughly 330 or so games which seems minuscule compared to the whole set but if the parent or clone runs from a set it probably a pretty good sign. What do you think would be a good title that you will be ok looking at months later. ;) Also what section "Ideas and Development"?

**Comment by markwkidd** [comment]
It's a classic moment when I suggest to someone else to start a retropie thread and then I break the retropie setup script.

@Wilstorm I see that @dankcushions put their mame2003 development thread in "Ideas and Development" so that does seem like the right spot.

**Comment by dankcushions** [comment]
the binary is when jools adds it to his buildbot. he won't do that until it leaves 'experimental' which is going to be hard ask until it builds successfully for a few days in a row ;) i wouldn't worry about it, given that we're in active development it will be out of date hours after the binaries are built, and they don't tend to get rebuilt until the next retropie release.

the libretro buildbot may have rpi builds though, maye?

**Comment by markwkidd** [comment]
I wonder if binaries from this section of the buildbot would work with an rpi. Anyone care to try? https://buildbot.libretro.com/nightly/linux/armhf/

**Comment by markwkidd** [comment]
@Wilstorm & others -- I don't have any more big refactoring or library updates on the agenda.

I'd like to eventually use more of the libretro-common/file_path functions to handle file paths, now that the library is available, but there's nothing coming from me soon that's especially likely to screw up the buildbot or RetroPie-Setup. Knock on wood.

Once the RetroPie PR is approved, or improved and replaced, then knock on wood, it should be a good time for more people to start using and testing the core.

**Comment by grant2258** [comment]
ive complies and installed mame2003 its fine had an error on the retroarch config that was a problem with mame itself though. Im compiling mame 2003 now ill let you know how i get on and if its all good

**Comment by Wilstorm** [comment]
> and they don't tend to get rebuilt until the next retropie release.

That makes sense now. I swear certain games worked for a long time and then didn't after a rebuild and I updated from binary and still a no go but after a source update all worked again. So that binary can be months old before being updated.

> I am going to learn to type one day.

Remember fingers on a s d f (left hand) and j k l ; (right hand)...you got this!

**Comment by grant2258** [comment]
ok mame2003 is building again when jools accepts that script we have a little problem though mark not a big one


‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/mame2003-plus_libretro.so’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/mame2003-plus_libretro.so’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/README.md’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/README.md’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/CHANGELOG.md’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/CHANGELOG.md’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/metadata’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/catver.ini’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/metadata/catver.ini’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/cheat.dat’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/metadata/cheat.dat’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/hiscore.dat’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/metadata/hiscore.dat’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/mame2003.xml’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/metadata/mame2003.xml’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/nvram’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/metadata/nvram’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/nvram/bubblem.nv’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/metadata/nvram/bubblem.nv’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/nvram/rungun.nv’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/metadata/nvram/rungun.nv’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/nvram’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/nvram’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/nvram/bubblem.nv’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/nvram/bubblem.nv’
‘/home/pi/RetroPie-Setup/tmp/build/lr-mame2003-plus/metadata/nvram/rungun.nv’ -> ‘/opt/retropie/libretrocores/lr-mame2003-plus/nvram/rungun.nv’
/home/pi/RetroPie-Setup

the scripst looking for the ram to be in /libretrocores/lr-mame2003-plus/nvram/rungun.nv’

there are in there the rest are in metadata when need then in one place. the script will need updating if the path changes

**Comment by grant2258** [comment]
@Wilstorm if you want to build the latest mame source for test you can download the script to retropie in retro pie with 
wet https://github.com/grant2258/mame2003-plus-libretro/raw/master/lr-mame2003.sh

chmod +x lr-mame2003.sh

this will fix mame and mame 2003

**Comment by Wilstorm** [comment]
Ok I grabbed the script. Do an update from source and run the script afterwards?

**Comment by grant2258** [comment]
you do the script first this is what builld sources uses

go this 

cd /home/pi/RetroPie-Setup/
wet https://github.com/grant2258/mame2003-plus-libretro/raw/master/lr-mame2003.sh
chmod +x lr-mame2003.sh
sudo cp lr-mame2003.sh scriptmodules/libretrocores/

then just install how you normally install your source

if you update the script youll need to do it again but it will get you compiling until the pull request is accepted or denied

**Comment by Wilstorm** [comment]
@grant2258 - Oh, nice, thank you for the instructions, that's really handy!

**Comment by Wilstorm** [comment]
Have the new changes been accepted or still waiting?

**Comment by markwkidd** [comment]
Still waiting. @Wilstorm you can subscribe to that PR on the right side of the screen at this link if you want an email when there is activity: https://github.com/RetroPie/RetroPie-Setup/pull/2375

**Comment by markwkidd** [comment]
@grant2258 the big update is hack CD soundtrack support for Final Fight, Mortal Kombat, and NBA Jam if you hadn't seen that.

**Comment by grant2258** [comment]
well that in itself justifies the update!

**Comment by Wilstorm** [comment]
Thank you and subscribed.

**Comment by grant2258** [comment]
@markwkidd mark there is still the issue with the nvram dir and one in metatdata as well the script is using nvram

**Comment by grant2258** [comment]
@markwkidd  ive fixed the script up made a few minor changes fro nvram if you could add these changed to you commit on retropie/ scripts at the same place on my repo.

**Comment by KMFDManic** [comment]
TEST Video of Final Fight with Sega CD OST, Mortal Kombat with Custom OST, and NBA Jam with Sega CD OST.  I already Pull Requested the Final Fight into MAME 2003 Plus.  I will Pull Request the other two, sometime today.  I mostly work with NES and SNES Classic.  But, everything works fine on the PC side of things, as well.

**SNES Classic - NBA Jam Arcade with SEGA-CD OST!!! The Trifecta of OST Perfection!!!**

https://www.youtube.com/watch?v=VW_ompmEsEE

I have all the necessary Samples Available, too.  They are all posted in my Core Set, in Xtras/Arcade Folder.  Thanks for having me involved in one of my personal favorite Cores:)

https://github.com/KMFDManic/NESC-SNESC-Modifications/releases

**Comment by markwkidd** [comment]
@grant2258 thanks for looking at the retropie script again -- I have submitted your updates to my retropie PR

@KMFDManic welcome welcome welcome!

**Comment by grant2258** [comment]
@KMFDManic Welcome aboard nice to see you here!

**Comment by markwkidd** [comment]
I think mame2003-plus is now good to go once users update RetroPie Setup

**Comment by grant2258** [comment]
cd /home/pi/Retropie-Setup
sudo git stash

**Comment by Wilstorm** [comment]
@grant2258 - Thank you good sir! :)

**Comment by grant2258** [comment]
mark he rejected it completely nothing changed

**Comment by grant2258** [comment]
@KMFDManic  i have two fixes one for mame2003-plus that does game timing properly

and i dont want to mess with mame2003 so the music plays a little faster when the fps is lower than 60 30 fps isint effected which one would you like?

**Comment by grant2258** [comment]
my latest one for mame2003 plus is here https://github.com/libretro/mame2003-plus-libretro/pull/94

**Comment by KMFDManic** [comment]
@grant2258 I can test both.  Main concern is retaining Data East and Irem encryption fixes, as well as the 3 Custom OSTs I added.  Much appreciated.  I know this was a challenge for awhile.  Whatever I need to add, code wise, to test, recommend away.  Hopefully, you had a chance to try our custom Final Fight Sega CD OST:)

**Comment by grant2258** [comment]
i will indeed i think i have the cd about somewhere but ill just download them the cd is probably unreadable after all this time. I just finished a shift at work will give it a whirl when i get up!

**Comment by grant2258** [comment]
if you want the right pitch and music speed on mame2003 dank said you just need to set the audio scew and restart retroarch

**Comment by grant2258** [comment]
if you count the music timing in robocop with mame2003 and mame2003 plus youll notice it straight away

**Comment by KMFDManic** [comment]
@grant2258 You can grab the samples from my Github release, Xtras/Arcade

I posted test video a few posts above.

So, to confirm, which commit should I add, so the Midway Bally will work?  Just dont want to muck things up on my end:)

**Comment by grant2258** [comment]
both work the mame2003 =+ one is the one i prefer at the moment

**Comment by KMFDManic** [comment]
@grant2258 Very interesting, timing differentials.  I did some manipulation to find good mixture of timing to help games like Elevator Action Returns run better on the SNESC.  I can see why the timing is such a sensitive subject in the Midway respect.  If you recommend the + timing fix, I can try that. Worst case scenario, I can put together a Midway Variant for us Mini Users, in the interum, til things are worked out. Shaders and other cpu intensive settings can potentially do wonders for offsetting fast games, as well.

**10 minute mark to see the Elevator Action Returns adjustment**

Game normally runs awful on both Retropie and NESC/SNESC.  The magic of exploiting RetroArch Settings:)

https://youtu.be/vVMznvqbezk

Looking forward to testing the Midway Timing!

**Comment by grant2258** [comment]
That is interesting. It seems like a lot of work! I'll need to test ea2 I know it's a cpu intensive game

**Comment by grant2258** [comment]
@KMFDManic it seems elevator action return is crashing on windows at least in mame2003+ now  i need to see if this is the core doing this sound changes that done this or something else we have changed it works in mame2003

**Comment by KMFDManic** [comment]
@grant2258 I tested the changes on my end.  Yes, it appears other games crash, possibly as a result of the changes.  Test out In the Hunt and R-Type LEO.  I would suspect several other Irem Games to crash as well.  Could be a conflict with the original encryption fixes.  There is another potential workaround, that I would suggest, if other games are affected.  Key in the rom set names, (ie: tron, rampage, etc) as triggers, and only allow the change for rom sets white listed within the context of that perimeter.  I did just this for my personal Build and had great success.  Also, I believe Battle Zone and Lunar Lander also have the timing issue, in general.  It is an awesome change, regardless.  I have applied it to many games and it is working excellent:)

**Comment by grant2258** [comment]
@KMFDManic ill fix this or change it thanks for the input. I just need to change the condtions a little if that doesnt work will do it the mame2003 way. Thanks for the info on the other games!

**Comment by KMFDManic** [comment]
No problem, @grant2258, I am very aware of the "Butterfly Effect"":)  But, check out Battle Zone and Lunar Lander, and any other games on the driver, potentially.  I, on my end, strictly white listed the Midway Bally Games, case by case, to effectively safeguard against any other breaks...at least for the time being:)  When I post my Update today, I will shout-out to you.  Thanks so much, again for the info on the timing input fixes:)  And, anything you need tested, let me know.

**Comment by grant2258** [comment]
@KMFDManic i think i know where i went wrong i got lazy and just done the sample rate * framerate this only needs applied it framerate is below 60k thanks for then mention!

**Comment by KMFDManic** [comment]
@grant2258 One thing I really want to be able to do.  And, this is a little pesky and time consuming.  But, I want to slowly but surely, in my readme files, instead of the generic label, Core by Libretro, I want to have Core by Libretro, Commits by: @markwkidd @grant2258 etc:)  I don't want this old, Atari (generic label with the true victors being overlooked scenario going on:)  I am very slowly but surely adding all the individuals who have worked on various commits for each of the amazingly respective Cores!  I have to say Twinaphex must never get any sleep with how often his name turns up, haha!

**Comment by grant2258** [comment]
@KMFDManic   Twinaphex, dankcushions and markwkidd are doing so much work you would think someone should be paying them!

**Comment by markwkidd** [comment]
@KMFDManic you're name is starting to join this list too!

Please submit PRs here any time. Again I'm glad to add some `ifdef`s to the source here for any modifications that you use that are platform-specific to the SNES Classic to make it easier to base your fork on this repository.

Also if you think you'd like to work directly in this repository it seems like you could probably get direct push access as well. Say something if this last idea sounds interesting to you!

**Comment by ZappaUtopia** [comment]
If you have to whitelist certain games to only use the timing fix there are quite a few. The timing fix also fixed the parent version of Green Beret (couldn't coin up). Cinematronics and Atari (probably all Vector games) were fixed.

I found another game that no longer works on Plus, `dungeonm` (Dungeon Magic). It works on 2003.

**Comment by arcadez** [comment]
This reminds me of when getting the sound to work in Denjin Makai it would break the
sound for Cabal, the classic rob peter to then pay paul in the end as @KMFDManic 
suggested the workaround was to just isolate the code so only DM used it.

/* Denjin Makai definitely needs this at start-up, it never writes to the bankswitch */
	{
	if(strcmp(Machine->gamedrv->name, "denjinmk") == 0)
        seibu_bank_w(1,0);
}

I suppose should we not be able to find general core timing values that all the games will like
there is always the option of tagging on a per game basis.

**Comment by KMFDManic** [comment]
@ZappaUtopia Thanks for the Green Beret mention.  @arcadez Yes, whitelisting can be a little time consuming.  But, essentially, trying to find a median for all games can have so many more possibilities and unexpected outcomes.  Imagine the one game that noone plays on MAME, for 5 years!  Then, all of a sudden, it comes up in an issue (Hey, why is this game not working, it worked 5 years ago!)  I tend to have a specific list of test perimeters, each time I do Updates (Irem Games most notably:)  Then, down the list with Data East, etc!  Butterfly Effect is awful when you change one thing for the better, and 10 things get broken in the process!

**Comment by grant2258** [comment]
I would prefer to not whitelist for now so any bugs show up

**Comment by KMFDManic** [comment]
@grant2258 This is just my own personal safety net, until tests work out.  Ideally, would like to be completely up to date with the main master, for sure!

**Comment by grant2258** [comment]
lemme try fix this mess up i made lol!

**Comment by ZappaUtopia** [comment]
I think the now working list is mainly Parent versions, the clones will probably work as well. Atari games are not on that list like Battlezone and Star Wars which previously didn't work but do now because of Grant's fix. Maybe All games below 47 Hz may now work that previously didn't regardless of Manufacturer, like Green Beret Parent version that I mentioned and also Big Run (but has graphical issues) and Cisco Heat.

**Comment by markwkidd** [comment]
If you do notice clones that are now working, it would be good to add them to the CHANGELOG as well. Folks are welcome to submit PRs or if you aren't comfortable doing that yet you can post an issue with what to add to the log.

Anything else that needs to get added to the CHANGELOG or README would be welcome as well.

**Comment by grant2258** [comment]
that will be the case zappa im having github problems again i had to just reset my repo will try get them fixes done just reset my github page

**Comment by grant2258** [comment]
just say any games below >47 fps now work like mame078 does you wont have to list all games that way

**Comment by ZappaUtopia** [comment]
I don't use a full collection because that's overkill ! I generally have Parent games only unless the Parent doesn't work or there is a better clone version (English version or Joystick hack etc.) I think it's safe to say that games below 47Hz that previously didn't work or wouldn't coin-up now do. But a few are now not working but could be caused by anything (dungeonm , elvactr). If you use a MAME 0.78 frontend you can order the Full gamelist by Refresh Rate and see which is below 47 Hz.

**Comment by grant2258** [comment]
ok guys its fixed was just a simpple one timing code is still good what romnames should i be testing ?

**Comment by grant2258** [comment]
@KMFDManic 

Thread 1 "retroarch" received signal SIGSEGV, Segmentation fault.
0x00007fffc61cab1e in IremGA20_update () from /home/grant/.config/retroarch/cores/mame2003_libretro.so
(gdb) bt
#0  0x00007fffc61cab1e in IremGA20_update () from /home/grant/.config/retroarch/cores/mame2003_libretro.so
#1  0x00007fffc5b8832a in streams_sh_update () from /home/grant/.config/retroarch/cores/mame2003_libretro.so
#2  0x00007fffc5b87efb in sound_update () from /home/grant/.config/retroarch/cores/mame2003_libretro.so
#3  0x00007fffc5b0d319 in updatescreen () from /home/grant/.config/retroarch/cores/mame2003_libretro.so
#4  0x00007fffc5b5f185 in cpu_vblankcallback () from /home/grant/.config/retroarch/cores/mame2003_libretro.so
#5  0x00007fffc5b6a2a0 in timer_adjust_global_time () from /home/grant/.config/retroarch/cores/mame2003_libretro.so
#6  0x00007fffc5b5f88a in mame_frame () from /home/grant/.config/retroarch/cores/mame2003_libretro.so
#7  0x00007fffc5b0acab in retro_run () from /home/grant/.config/retroarch/cores/mame2003_libretro.so
#8  0x00005555555836cc in core_run () at core_impl.c:430
#9  0x0000555555588f94 in runloop_iterate (sleep_ms=0x7fffffffe090) at retroarch.c:3261
#10 0x000055555558212d in rarch_main (argc=1, argv=0x7fffffffe1a8, data=0x0) at frontend/frontend.c:131
#11 0x000055555558218a in main (argc=1, argv=0x7fffffffe1a8) at frontend/frontend.c:154


did rtypeleo ever work with mame2003?

**Comment by grant2258** [comment]
@ZappaUtopia  @KMFDManic irem games fixed thanks for info that the games had problems

**Comment by grant2258** [comment]
@markwkidd  is there a easy way for me to pass this commit for irem to mame2003 as a pull req without redoing it for there. I need to leave for work will pass it alone to mame2003 when i get back in

**Comment by grant2258** [comment]
@markwkidd  thanks for that . I think i would need access to mame2003 to do that ill just do it the old fashioned way. I swear i spend more time on git hub than programming lol

**Comment by markwkidd** [comment]
You don't need admin access to mame2003 for this process -- which I do not have either -- but you would be able to go over to github at that point and generate a PR to mame2003 that someone with admin access could merge.

Yes code management is its own hobby I've learned

**Comment by ZappaUtopia** [comment]
@grant2258 I haven't built yet but did this also fix `dungeonm` (Dungeon Magic by Taito) ? It works on 2003 but not on 2003-plus.

**Comment by grant2258** [comment]
@ZappaUtopia yes that game works on mame2003+

**Comment by markwkidd** [comment]
I now have things so that the nvram 'bootstrap' patch files aren't even needed for `rungun` and `bubblem`. It's integrated into the core.

At least it's working that way here in Windows. If anyone wants to test, update mame2003-plus, delete any nvram files you already have in your save folder, and then start up `rungun` or `bubblem`.

**Comment by grant2258** [comment]
@ZappaUtopia  the fixes for irem or for 64 bit in particular. I wouldnt have even know about this unless someone told me there was issues. The reason there was problems on some game is because mark done some massive core work and he did a great job. We both done changes at the same time on a particular file and one line of code didnt make the cut its added back now.

**Comment by KMFDManic** [comment]
@markwkidd Awesome work on Rungun!  Can't wait to check it out!

**Comment by KMFDManic** [comment]
@markwkidd @grant2258 @twinaphex Latest Addition to MAME 2003 Xtreme, on my end:)

**SNES Classic - Out Run Arcade - Custom OST Support - Test Two Redux)**

https://www.youtube.com/watch?v=G-YO7nJMKqE

**Comment by markwkidd** [comment]
These videos are so great so we can all actually see what's being
described! Plus the music is rockin!

**Comment by KMFDManic** [comment]
@markwkidd @grant2258 @dankcushions I will be adding MAME 2003 Plus in Today's Update Release for the NES/SNES Classic Users.  My preliminary ReadMe! will be as follows

```-----------------------
Name: km_MAME 2003 Plus Experimental (4-30-18)
Creator: KMFDManic
Category: KMFDManic Experimental Core Set
-----------------------
=== MAME2003 Plus Experimental Core for RetroArch ===

Module adds support for MAME 2003 Plus Experimental

Bios must be inside game directory or via Bios Module where applicable

This Particular MAME Plus Experimental Variant offers the most current experimental
changes to MAME 2003. While generally stable in a PC Environment, you 
may be subject to various incompatibility issues in an NES/SNES Mini
Environment. Of course, User Feedback will be taken into account for Updates
on a near daily basis to further stabilize the Core for compatibility across
many different Platforms!:)    

Available executables and arguments to run Core:
- /bin/mame2003-plus <rom> <clover_args>

Core by libretro

libretro commits by: markwkidd/arcadez/dankcushions/twinaphex/hunterk/grant2258/gamez-fan/etc

Hakchi module system by madmonkey

NES/SNES Mini shell integration by Cluster

(c) 2016-2018
```

If any of you have anything to add to the ReadMe!, please let me know:)  I will need to keep two MAME 2003s in my Core Set.  Xtreme will be changes that are stable and work on the Minis with reliable functionality.  Then, myself and many others will continue to test any and all changes to MAME 2003 Plus, and I will fill everyone in, accordingly:)  Thanks so much for all of the truly amazing Commits!  I will be sure to notate a few of them in the Release Notes, today!

**Comment by fr500** [comment]
I still don't know what you change for nes/snes mini and why those changes aren't pull requested to our repos

**Comment by KMFDManic** [comment]
@fr500 As I mentioned before, it is a WIP, and the integral part of it all is the program we use called Hakchi2 CE.  RetroArch on PC/Android handles the Cores.  Hakchi2 CE on Mini NES/SNES, is absolutely necessary to be able to use RetroArch on the Minis.  There is quite a lot of developing going on.  We just got WI-FI working on our end!  But, in the end, the changes that are helpful, globally, across platforms...will be Pull Requested.  The Custom OST ones, and a few other things I have noticed, have already been brought to the attention of @markwkidd @grant2258 and @twinaphex 

The RetroKeyboard is fantastic, by the way:)  

Now that I am in constant communication with those who work on MAME 2003 Plus, I will relay any information helpful to furthering the whole cause:)  Thanks again!

**Comment by fr500** [comment]
The RetroArch and core changes have to be pushed back (or at least offered back) no matter if it's a WIP or not.

**Comment by markwkidd** [comment]
I have heavily rewritten the README file with new users in mind: https://github.com/libretro/mame2003-plus-libretro/blob/master/README.md

**Comment by markwkidd** [comment]
done!
